internal/controller/drpolicy_controller.go

```diff
@@ -295,79 +326,46 @@ func hasConflictingDRPolicy(match *ramen.DRPolicy, drclusters *ramen.DRClusterLi
 		}
 
 		// None of the common managed clusters should belong to Metro Regions in either of the drpolicies.
-		if haveOverlappingMetroZones(match, drp, drclusters) {
+		if haveOverlappingMetroZones(match, drp, drclusters, drClusterIDToName) {
 			return fmt.Errorf("drpolicy: %v has overlapping metro region with another drpolicy %v", match.Name, drp.Name)
 		}
 	}
 
 	return nil
 }
 
-func getClusterIDSFromPolicy(drpolicy *ramen.DRPolicy) []string {
-	cids := []string{}
-
-	if len(drpolicy.Status.Sync.PeerClasses) > 0 {
-		for _, pc := range drpolicy.Status.Sync.PeerClasses {
-			cids = append(cids, pc.ClusterIDs...)
-		}
-
-		return cids
-	}
-
-	if len(drpolicy.Status.Async.PeerClasses) > 0 {
-		for _, pc := range drpolicy.Status.Async.PeerClasses {
-			cids = append(cids, pc.ClusterIDs...)
-		}
-
-		return cids
-	}
-
-	return cids
-}
-
-func hasOverlapWithMetro(metroMap map[string][]string, commonClusterIDs, commonClusters []string) bool {
-	for _, v := range metroMap {
-		if sets.NewString(v...).HasAny(commonClusterIDs...) {
-			return true
-		}
-
-		if sets.NewString(v...).HasAny(commonClusters...) {
-			return true
-		}
-	}
-
-	return false
-}
-
-func haveOverlappingMetroZones(d1 *ramen.DRPolicy, d2 *ramen.DRPolicy, drclusters *ramen.DRClusterList) bool {
+func haveOverlappingMetroZones(
+	d1, d2 *ramen.DRPolicy,
+	drclusters *ramen.DRClusterList,
+	drClusterIDToName map[string]string,
+) bool {
 	d1ClusterNames := sets.NewString(util.DRPolicyClusterNames(d1)...)
-	d1ClusterIDs := sets.NewString(getClusterIDSFromPolicy(d1)...)
-	d1SupportsMetro, d1MetroMap := dRPolicySupportsMetro(d1, drclusters.Items)
-
+	d1SupportsMetro, d1MetroRegions := dRPolicySupportsMetro(d1, drclusters.Items, drClusterIDToName)
 	d2ClusterNames := sets.NewString(util.DRPolicyClusterNames(d2)...)
-	d2ClusterIDs := sets.NewString(getClusterIDSFromPolicy(d2)...)
-	d2SupportsMetro, d2MetroMap := dRPolicySupportsMetro(d2, drclusters.Items)
-
+	d2SupportsMetro, d2MetroRegions := dRPolicySupportsMetro(d2, drclusters.Items, drClusterIDToName)
```

Comment:
what about renaming `d2MetroRegions` to `d2MetroMap` since we are not returning region anymore.  The same goes to `d1MetroRegions`